### ISSUE TITLE
[REQ] Tracker Request - https://www.1tamilmv.pw/

### ISSUE THREAD

**Issue by drkathirpi** [Jackett/Jackett#12097]
contains lot of content for tamil language. kindly add it to jackett will help a lot for tamilians all over the world

**Comment by garfield69** [comment]
public forum-style site. it has no standard torrent listing search page
I think this:
https://www.1tamilmv.one/index.php?/search/&q=loki&type=forums_topic&search_in=titles
is about as close as we can get to a regular torrent listing search page, we might be able to get the size off the title, but it does not have the usual seeders, leechers, flies, grabs, posters, etc

forum posts can however have many torrents for one topic, (such as the loki season one with 6 episode torrents) so this site will not be suitable for coding with a yaml indexer, it will require one written in C#.
as such I expect a very long wait time as c# dev volunteers for new c# indexers are in short supply :-(

**Comment by mikeoscar2006** [comment]
Two questions if you don't mind answering @garfield69 which I've seen common in most open requests for trackers:
1. What is the correct way to deal with such listings where each title has many posts and each post has multiple torrents. I mean what should be the expected outcome?
2. Most listings only give you the title and date and hence most info for a "Release" is missing and even keeping that aside, the size is even not available. Should the values be blank or random and does it even look good?

**Comment by garfield69** [comment]
>What is the correct way to deal with such listings where each title has many posts and each post has multiple torrents. I mean what should be the expected outcome?

In the early days of Jackett, the approach would have been to just fetch the websites' page of results, and for each result row fetch the corresponding details page and use what ever info was there to create a jackett search results for each torrent found, building rows from the combined available data.
I believe we still have several C# indexers that use this method.
However, now days, with the heightened awareness of DDoS and the web sites sensitivity to bot attacks, I am hesitant to endorse such a brute force method to gather torrent info, as we are potentially looking at generating up to 101 http requests for pages on a web site in a short amount of time, whenever a search is performed.
So I would first look to see if the web site provides an API (JSON or XML) and use that if its is available.
If that is not available, then before committing to a fetch-details-pages indexer, I would first consider what info is available on the details pages.
If the site does not provide at least the category, size, seeders, leechers, and upload date, then I am inclined to flag the site as not suitable for use with Jackett.
The site must also provide either a .torrent URL or a magnet URI directly (Torznab requirement) and not via 3rd party redirects (as those sites may be blocked by ISP even if the indexers site is not), and direct play streaming downloads and filesharing direct file downloads are not supported.

>Most listings only give you the title and date and hence most of the info for a "Release" is missing and even keeping that aside, the size is even not available. Should the values be blank or random and does it even look good?

If the site has the occasional size missing, we normally would default it to **0B**. If however the site never provides a size, then we would provide a default of some sort, as returning 0B for everything can cause some torznab apps to reject the torrents. 
For some we default everything to **512MB**, and for others, if there is a clear consistent category structure, then we can provide category defaults, for example:
5GB for movies, 512MB for TV and music, 512KB for e-books etc, basically an average of what ever size is most prevalent at the site.
Defaults for the occasional missing seed and leech is **0**, but if seeders and leechers are always missing on the site then the default is usually **1**, as once again some torznab apps will reject torrents with 0 seeders.
Files and grabs can remain undefined as this is quite common (I mean left out of the release definition altogether, and not defined and set to null or blank).
Date can usually be set to **now** if it is not available.


>Many public trackers have separate categories (even for resolutions) and torrents are well sorted in each of them but the tracker doesn't allow search in individual trackers (i.e. site-wide only). Does it even make sense to add separate Category Mappings because there won't be any way to search them.

We try to tag a category for each torrent otherwise torznab apps will not accept torrents. If there are no categories available at all, then we default everything to OTHER. If the site supplies no categories, but the torrents are of a particular type (TV, MOVIES, BOOKS, GAMES) then we add a catmap for that category in addition to OTHER, as the corresponding search mode to the indexer, so that torznab apps will accept the indexer and can be made to accept the torrents provided.
Even if the site does not provide category filtering on their search page, we try to provide the categories if they are available, so that the torrents are tagged and torznab apps can decide to accept to reject them.

**Comment by mikeoscar2006** [comment]
Thanks for answering the question, will take some time and go through it before implementing the indexers.

**Comment by mikeoscar2006** [comment]
One last question which was not answered above @garfield69, What if one title has multiple magnet links on the details page, not separated by any kind of release just varying in size (example javbus.com) or separated by release but again, we only get to know at the details page (eg. pasateatorrent.net)? Do we select a random magnet or with the largest size if available or an option for config which allows user to select whether random or largest?

**Comment by garfield69** [comment]
No, the point is to provide all the related .torrents or magnets in the details page associated with the initial poster in the search page.
So for these, if the title in the magnet itself is not enough to tell them apart then I would add the magnet's hash at the end of the title, and extract as much other details for each row from what is available in the details page for each magnet (size, dates, etc).

**Comment by mikeoscar2006** [comment]
But what you said earlier was correct that we won't be making 101 requests to parse everything. So the details page will only be parsed when let's say the user clicks on the Download button right? We can't add more results to the search as at that point we just have to provide a download link.
So let's say in javbus, we search for "SNMD" and it starts showing different posters which will be shown when the search is pressed, but when the user presses the Download button next to let's say SNMD-1712 title, its details page will be parsed and it's going to have let's say 5 magnet links with the same name in lowercase eg, "snmd-1712.mp4" and different sizes. So here we have to choose one only right? So that's what I was asking which one to choose? or is there any other way you're trying to convey?

**Comment by garfield69** [comment]
Well now you understand why these one-title-many-torrents sites are such a pain in the ass. ;-)

Take a web site that has a single poster in their search page for a tv series season, and  its details page has each episode in that series listed with its own magnet.
If we are not going to fetch the details page while building the results rows to provide each episode as a row, then there is little point providing an  indexer for that site.

And if for instance, we are going to fetch the first magnet in the details page at download time, then we might as well just write the indexer in yaml because we don't need a C# indexer for that.

So yeah, I don't see there being much benefit in providing a C# indexer that will only return one torrent, as we have no way to determine which torrent is the one the user would prefer to have.

So I suppose the only way forward is to  go ahead and fetch the details pages for each search result title, and at the first sign that the site starts to flag this indexer as an unwanted traffic generator we remove the indexer and call it a day.

OR we just close those tickets as unworkable in the current DDoS awareness climate.

**Comment by mikeoscar2006** [comment]
> Well now you understand why these one-title-many-torrents sites are such a pain in the ass. ;-)
Yeah, you can count me in :)
You're right, we can at least create the indexers and later put them on hold or find a way to regulate traffic. 
I'll start implementing trackers first thing in the morning and Thank you so much for bearing with my mountain of questions.

**Comment by garfield69** [comment]
Closing as unsuitable for indexing. Refer https://github.com/Jackett/Jackett/wiki/How-to-request-a-new-tracker#what-makes-a-site-suitable-or-not-suitable-for-indexing

**Comment by KarthiSurfer** [comment]
mediafusion stremio addon indexes this site for content, the code is aldready there you can just use it to implement this - https://github.com/mhdzumair/MediaFusion
